### PR TITLE
Convert legacy retryable reads tests to unified format

### PR DESCRIPTION
**Pull request by jyemin**
JAVA-5344

### CODE REVIEW

File: driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedRetryableReadsTest.java [hunk 1]
```diff
@@ -24,12 +24,22 @@
...
-                                      final String schemaVersion, final BsonArray runOnRequirements, final BsonArray entitiesArray,
-                                      final BsonArray initialData, final BsonDocument definition) {
+    public UnifiedRetryableReadsTest(final String fileDescription, final String testDescription, final String schemaVersion,
+            final BsonArray runOnRequirements, final BsonArray entitiesArray, final BsonArray initialData, final BsonDocument definition) {
         super(schemaVersion, runOnRequirements, entitiesArray, initialData, definition);
+        customSkips(fileDescription, testDescription);
+    }
+
+    public static void customSkips(final String fileDescription, @SuppressWarnings("unused") final String testDescription) {
+        assumeFalse(fileDescription.equals("count"));
```
Comment (rozza) [inline driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedRetryableReadsTest.java:37]: Perhaps add a comment why these are skipped.

File: driver-scala/src/integration/scala/org/mongodb/scala/RetryableReadsTest.scala [hunk 2]
```diff
@@ -1,44 +0,0 @@
...
- * limitations under the License.
- */
-
-package org.mongodb.scala
-
-import com.mongodb.client.AbstractRetryableReadsTest
-import org.bson.{ BsonArray, BsonDocument, BsonString }
-import org.mongodb.scala.syncadapter.SyncMongoClient
-
-class RetryableReadsTests(
```
Comment (rozza) [inline driver-scala/src/integration/scala/org/mongodb/scala/RetryableReadsTest.scala:23]: Is there a unified test for Scala now?

File: driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedRetryableReadsTest.java (hunk 1 above)
Comment (jyemin) [reply driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedRetryableReadsTest.java:38]: Done

File: driver-scala/src/integration/scala/org/mongodb/scala/RetryableReadsTest.scala (hunk 2 above)
Comment (jyemin) [reply driver-scala/src/integration/scala/org/mongodb/scala/RetryableReadsTest.scala:23]: No.  Given that Kotlin, Kotlin co-routine, and Scala are all thin wrappers around the Java-based drivers, I don't think it's worth it to test retryable reads against all of them, as all the logic that's being tested lives in driver-core and is well-tested by the test runners for sync and reactive drivers.

Comparing with the other unified tests, I see only UnifiedCrudTest for kotlin-coroutine and kotlin-sync, so this seems in line with previous decisions.

**Review by vbabanin — APPROVED** [review]
LGTM!

**Review by rozza — APPROVED** [review]
LGTM

**Pull request merged**